vortex-array/src/array/struct_/mod.rs

```diff
@@ -150,11 +150,24 @@ impl VariantsVTable<StructArray> for StructEncoding {
 
 impl StructArrayTrait for StructArray {
     fn maybe_null_field_by_idx(&self, idx: usize) -> Option<ArrayData> {
-        self.dtypes().get(idx).map(|dtype| {
```

Comment:
the key thing is that `self.dtypes()` runs `field_types.iter().collect::<Vec<_>>()` to support lazily read DTypes.

This method is called surprisingly often! Most particularly, in compression when you're counting `nbytes()` all the time. On the test file in #1749 this accounted for a giant portion of the compress time with the 14k columns 1000 rows example file